doc/docs/Python_Tutorials/Cylindrical_Coordinates.md

```diff
@@ -455,6 +455,156 @@ As shown below, the results for the scattering cross section computed using cyli
+at each $m$ values are embarrassingly parallel so they can be run simultaneously.
+
+On the other hand, because the source amplitudes $J_r(r,m)$ and $J_\phi(r,m)$ are generally not constant and extend to infinity, we used the principle of equivalence (for reference, see [Electromagnetic wave source condition](https://arxiv.org/pdf/1301.5366.pdf)) to create equivalent sources that are of finite sizes.
+
+We present an example below that calculates the scattered flux of a sphere. Because of the spherical symmetry, incidence at different angle should have identical results. We can thus use this feature to check our approach. Note that because of the axial symmetry in the cylindrical coordinates, we cannot distinguish different azimuthal angles but we can distinguish different polar angles. We thus simply choose our incidence to be of form $E_ye^{ik_xx}$, and we can vary the angle of incidence by varying $k_x$.
+
+```py
+import numpy as np
+from scipy import special
+import meep as mp
+mp.verbosity(0)
+r = 0.6  # size of flux box
+cyl_r = 0.5 # radius of sphere
+h = 2 * r  # height/diameter of sphere
+
+wvl = 2 * np.pi * cyl_r / 4
+frq_cen = 1 / wvl
+dfrq = 0.2
+nfrq = 1
+resolution, mrange = 50, 5
+dpml = 0.5 * wvl
+dair = 1.0 * wvl
+pml_layers = [mp.PML(thickness=dpml)]
+sr = r + dair + dpml
+sz = dpml + dair + h + dair + dpml
+cell_size = mp.Vector3(sr, 0, sz)
+n_cyl = 2.0
+geometry = [mp.Sphere(material=mp.Medium(index=n_cyl), center=mp.Vector3(), radius=cyl_r)]
+
+k_cen = 2 * np.pi * frq_cen
+alpha_list = [0, np.pi/36, np.pi/24, np.pi/18, np.pi/12]
+alpha_range = len(alpha_list)
+
+
+src_size_tb = 2*r
+src_size_side = 3*r
+src_center_top = mp.Vector3(src_size_tb/2, 0, src_size_side/2)
+src_center_bottom = mp.Vector3(src_size_tb/2, 0, -src_size_side/2)
+src_center_side = mp.Vector3(src_size_tb, 0, 0)
+
+scatt_flux_m = np.zeros((alpha_range, mrange+1))
+for alpha_i in range(alpha_range):
+    alpha = alpha_list[alpha_i]
+    kxy, kz = k_cen*np.sin(alpha), k_cen * np.cos(alpha)
+    amp_side = lambda v3: np.exp(1j * kz*(v3.z+src_size_side/2))
+    phase_top = amp_side(src_center_top)
+
+    for cur_m in range(0, mrange+1):
+        if alpha!=0 or cur_m == 1:
+            coeff_p1 = 0.5 * (1j)**(cur_m+1)
+            coeff_m1 = 0.5 * (1j)**(cur_m-1)
+
+            src_cen = src_size_tb/2
+            Jpm = lambda v3: coeff_p1 * special.jv(cur_m+1, kxy * (v3.x+src_cen)) + coeff_m1 * special.jv(cur_m-1, kxy * (v3.x+src_cen))
+            Jrm = lambda v3: 1j * coeff_p1 * special.jv(cur_m+1, kxy * (v3.x+src_cen)) - 1j * coeff_m1 * special.jv(cur_m-1, kxy * (v3.x+src_cen))
+            Jside = (1j)**cur_m * special.jv(cur_m, kxy*src_size_tb) * kxy/k_cen
+
+
```

Comment:
Would be helpful to add a comment to describe the setup of the equivalent sources based on the equations derived in the reference from Zin Lin.